### PR TITLE
Feature/get books

### PR DESCRIPTION
**Pull request by stdroniv**
Implement get books enpoint, add basic query pagination, update books image_url column name, add few new methods for generating openapi specs

### CODE REVIEW

File: internal/httputil/pagination.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+		if pn < 0 || err != nil {
+			return nil, errors.New("invalid page number. value should not be less than 0")
+		}
+
+		pagination.PageNumber = pn
+	}
+
+	pageSize := qs.Get("size")
+	if pageSize != "" {
```
Comment (obaibula) [inline internal/httputil/pagination.go:39]: This if clause is redundant. You get an error from strconv.Atoi if you pass empty string into it.

File: internal/httputil/pagination.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+func ParsePaginationQuery(r *http.Request) (*PaginationData, error) {
+	qs := r.URL.Query()
+
+	pagination := &PaginationData{
+		PageNumber: 0,
+		PageSize:   10,
+	}
+
+	pageNumber := qs.Get("page")
+	if pageNumber != "" {
```
Comment (obaibula) [inline internal/httputil/pagination.go:28]: This if clause is redundant. You get an error from strconv.Atoi if you pass empty string into it.

File: internal/assert/assertions.go [hunk 3]
```diff
@@ -69,6 +69,13 @@ func NotNil(t *testing.T, ptr any) {
 	}
 }
 
+func Nil(t *testing.T, ptr any) {
```
Comment (obaibula) [inline internal/assert/assertions.go:72]: This function has no usages.

File: internal/data/books.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+func (r *BookRepository) GetBooks(pd *httputil.PaginationData) ([]Book, int, error) {
+	query := `
+select b.id, b.title, b.image_urls, b.avg_review, b.created_at, b.updated_at
+from books as b
+offset $1
+fetch first $2 rows only
+	`
+	countQuery := `
+select count(id) from books
```
Comment (obaibula) [inline internal/data/books.go:41]: In postgresql count(*) is much faster than count(id)

File: internal/data/books.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	rows, err := results.Query()
+	if err != nil {
+		return nil, -1, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var b Book
+
+		err := rows.Scan(
```
Comment (obaibula) [inline internal/data/books.go:64]: Unshadow err: "err = rows.Scan("

File: internal/data/books.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+		books = append(books, b)
+	}
+
+	var booksCount int
+	err = results.QueryRow().Scan(&booksCount)
+	if err != nil {
+		return nil, -1, err
+	}
+
+	if err := results.Close(); err != nil {
```
Comment (obaibula) [inline internal/data/books.go:85]: Unshadow err "err = results.Close()"

File: internal/doc/books.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	successSchema.AddProperty("max_pages", "integer", "Maximum number of pages with a specified page size", true)
+
+	refSchema := NewRefSchema("Book")
+	successSchema.AddSchemaArrayProperty("data", "Paginated array of books", refSchema, true)
+
+	requestErrorSchema := NewSchema()
+	requestErrorSchema.AddProperty("errMsg", "string", "Specific error", true)
+
+	op.AddRequestSchema(reqSchema, "application/json", true)
+	op.AddResponseSchema(successSchema, "201", "application/json", "Success")
```
Comment (obaibula) [inline internal/doc/books.go:25]: It should be 200.

File: internal/httputil/pagination.go (hunk 1 above)
Comment (stdroniv) [reply internal/httputil/pagination.go:39]: this if clause makes pageSize query parameter required, I'm not sure if that's our desired behavior

**Pull request closed without merging**